apiserver error: first path segment in URL cannot contain colon

Hi,

I have been trying to run api-server as an standalone service along with etcd. I am passing etcd-servers and cluster-ip-range as arguments, here is the error I am getting:

 ```
./_output/bin/kube-apiserver --etcd-servers=http://127.0.0.1:2379 --service-cluster-ip-range=10.0.0.0/16
I1207 01:29:41.175437   16008 config.go:514] Will report 192.168.122.69 as public IP address.
W1207 01:29:41.179097   16008 config_selfclient.go:39] Failed to create secure local client, falling back to insecure local connection: <nil>
E1207 01:29:41.179125   16008 server.go:245] Failed to create clientset: parse 127.0.0.1:8080: first path segment in URL cannot contain colon
panic: parse 127.0.0.1:8080: first path segment in URL cannot contain colon

goroutine 1 [running]:
k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion.NewForConfigOrDie(0xc420756700, 0xc42072adf0)
    /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion/core_client.go:132 +0x68
k8s.io/kubernetes/pkg/master.(*Config).Complete(0xc42072bce8, 0x0)
    /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/master/master.go:181 +0x330
k8s.io/kubernetes/cmd/kube-apiserver/app.Run(0xc420580140, 0x24ca4f0, 0x5115440)
    /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:311 +0x16ac
main.main()
    /root/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kube-apiserver/apiserver.go:50 +0xc0
```

etcd servers are up and running on that address. Any leads as to how to resolve this? TIA.

--- a/pkg/client/restclient/url_utils.go
+++ b/pkg/client/restclient/url_utils.go
@@ -33,10 +33,7 @@ func DefaultServerURL(host, apiPath string, groupVersion schema.GroupVersion, de
 	}
 	base := host
 	hostURL, err := url.Parse(base)
-	if err != nil {
-		return nil, "", err
-	}
-	if hostURL.Scheme == "" || hostURL.Host == "" {
+	if err != nil || hostURL.Scheme == "" || hostURL.Host == "" {
 		scheme := "http://"
 		if defaultTLS {
 			scheme = "https://"

